openstf/model_selection/model_selection.py

```diff
@@ -1,87 +1,254 @@
-    test_fraction: float = 0.0,
+    data_: pd.DataFrame,
+    test_fraction: float = 0.1,
     validation_fraction: float = 0.15,
-    backtest: bool = False,
-    period_sampling: bool = True,
-    period_timedelta: timedelta = timedelta(days=2),
-) -> Tuple[pd.DataFrame, pd.DataFrame, pd.DataFrame]:
-    """Split input data into train, test and validation set.
+    back_test: bool = False,
+    stratification: bool = True,
+) -> (pd.DataFrame, pd.DataFrame, pd.DataFrame):
+    """
+    Split input data into train, test and validation set.
 
-    Function for splitting cleaned data with features in a train, test and
-    validation dataset. In an operational setting the folowing sequence is
-    returned:
+    Function for splitting data with features in a train, test and
+    validation dataset. In an operational setting the following sequence is
+    returned (when using stratification):
 
-    Test >> Validation >> Train
+    Test >> Train >> Validation
 
-    For a back test (indicated with argument "back_test") the folowing sequence
+    For a back test (indicated with argument "back_test") the following sequence
     is returned:
 
-    Validation >> Train >> Test
+    Train >> Validation >> Test
 
     The ratios of the different types can be set with test_fraction and
     validation fraction.
 
     Args:
-        data (pandas.DataFrame): Clean data with features
+        data_ (pandas.DataFrame): Cleaned data with features
         test_fraction (float): Number between 0 and 1 that indicates the desired
             fraction of test data.
         validation_fraction (float): Number between 0 and 1 that indicates the
             desired fraction of validation data.
-        backtest (bool): Indicates if data is intended for a back test.
-        period_sampling (bool): Indicates if validation data must be sampled as
-            periods.
-        period_timedelta (datetime.timedelta): Duration of the periods that will
-            be sampled as validation data. Only used for period_sampling=True.
+        back_test (bool): Indicates if data is intended for a back test.
+        stratification (bool): Indicates if validation data must be sampled as
+            periods, using stratification
 
     Returns:
-        Tuple with train data, validation data and test data:
-            [0] (pandas.DataFrame): Train data
-            [1] (pandas.DataFrame): Validation data
-            [2] (pandas.DataFrame): Test data
+        train_data (pandas.DataFrame): Train data.
+        validation_data (pandas.DataFrame): Validation data.
+        test_data (pandas.DataFrame): Test data.
+
     """
-    MIN_TRAIN_FRACTION = 0.5
-    logger = structlog.get_logger(__name__)
 
     # Check input
-    train_fraction = 1 - (test_fraction + validation_fraction)
+    data_["timestamp"] = data_.index
+    if "Horizon" in data_.columns:
+        data = data_[data_["Horizon"] == 47]
+    else:
+        data = data_
+        stratification = False
```

Comment:
This looks sketchy. I'm not sure what the functionality of these lines of codes is. Perhaps we can have a call and see if we can make it more clear or generalize it in some way?